python/pyiceberg/types.py

```diff
@@ -167,44 +212,45 @@ class DecimalType(PrimitiveType):
         True
     """
 
-    __root__: str = Field()
+    root: Tuple[int, int]
```

Comment:
Similar here, it's odd to keep the type state as "root" unless this is something pydantic requires.

Reply:
This is required. Because this doesn't (de)serializes to dict, it has to map onto the root.